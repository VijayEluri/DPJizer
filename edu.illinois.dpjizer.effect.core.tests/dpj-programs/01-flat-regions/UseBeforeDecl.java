/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package flatregion.usebefore;

/**
 * 
 * @author Mohsen Vakilian
 * 
 */
class C {
	region r1;
	MyPair p in r1;

	C() {}
	
	void m() {
		p.y = 5;
	}
}

class MyPair {
	int x;
	int y;
	
	MyPair() {}
}