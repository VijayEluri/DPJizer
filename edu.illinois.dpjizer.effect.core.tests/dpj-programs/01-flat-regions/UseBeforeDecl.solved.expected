flatregion.usebefore.C.C() pure
flatregion.usebefore.C.m() reads Root : flatregion.usebefore.C.r1 writes Root
flatregion.usebefore.MyPair.MyPair() pure
